File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//function's protoypes
int count_letters(string text);
int count_words(string text);
int sentence_count(string text);

int main(void)
{
    //gets user input
    string n;
    n = get_string("Text: ");
    //gets lengths for letters, words and sentences an assignes to corresponding integer variables
    float letter_length = count_letters(n);
    float word_length = count_words(n);
    float sentence_length = sentence_count(n);
    //calculates Coleman-Liau index
    float L = (letter_length / word_length) * 100;
    float S = (sentence_length / word_length) * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    int grade = round(index);
    //prints grade according to conditions
    if (grade < 1)
    {
        printf("Before Grade 1");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %d", grade);
    }
    printf("\n");
}

int count_letters(string text)
{
    int count = 0;
    int len = strlen(text);
    //loop to count how many alphabetic characters there are in a string
    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    int len = strlen(text);
    //loop to count how many white spaces there are in a sentence, which indicates a new word
    for (int i = 0; i < len; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    //returns count + 1 due to the fact that there is always a word after the last whitespace
    return count + 1;
}

int sentence_count(string text)
{
    int count = 0;
    int len = strlen(text);
    //loop to count how many period, question marks or exclamation marks there are, indicating a new sentence.
    for (int i = 0; i < len; i++)
    {
        if (strchr(".?!", text[i]))
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():

    string = get_string('Text: ')

    # Assigns various counts to variables
    letter_count = letters(string)
    word_count = words(string)
    sentence_count = sentences(string)

    # Calculates average letters and sentences per 100 words
    L = (letter_count/word_count) * 100
    S = (sentence_count/word_count) * 100

    # Calculates grade
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    # Returns grade based on score
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade {}".format(grade))

# Counts number of letters in text


def letters(string):

    count = 0

    for i in range(0, len(string)):
        if string[i].isalpha():
            count += 1

    return count

# Counts number of words in text


def words(string):

    word_counter = len(string.split(" "))

    return word_counter

# Counts number of sentences in text


def sentences(string):

    count = 0

    for i in range(0, len(string)):
        if string[i] == '.' or string[i] == '!' or string[i] == '?':
            count += 1

    return count


main()

